src/ssl.c

```diff
@@ -17838,123 +17839,50 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
-            if (wc_InitMutex(&rand_methods_mutex) != 0) {
-                WOLFSSL_MSG("Bad Init Mutex rand methods");
-                ret = BAD_MUTEX_E;
-            }
-            else {
-                initRandMethMutex = 1;
-            }
+    #ifndef WOLFSSL_NO_OPENSSL_RAND_CB
+        if (wolfSSL_RAND_InitMutex() == 0 && wc_LockMutex(&gRandMethodMutex) == 0) {
+            gRandMethods = methods;
+            wc_UnLockMutex(&gRandMethodMutex);
         }
-
-        if (initRandMethMutex != 0) {
-            if (wc_LockMutex(&rand_methods_mutex) != 0) {
-                WOLFSSL_MSG("Lock rand methods mutex failed");
-                ret = BAD_MUTEX_E;
-            }
-            else {
-                rand_methods = methods;
-                wc_UnLockMutex(&rand_methods_mutex);
-                ret = WOLFSSL_SUCCESS;
-            }
-        }
-
-        return ret;
-    }
-
-    /* Sets func_ptr to a pointer to the given function func if it was set
-       by RAND_set_rand_method, or NULL if either it was not set,
-       or there was an error retreiving it.
-
-       Returns WOLFSSL_SUCCESS on success, otherwise returns an error code. */
-    static int GetRandFunction(RandFunction func, void** func_ptr)
-    {
-        int ret = WOLFSSL_FAILURE;
-
-        if (func_ptr == NULL) {
-            return WOLFSSL_FAILURE;
-        }
-
-        *func_ptr = NULL;
-
-        if (initRandMethMutex == 0) {
-            if (wc_InitMutex(&rand_methods_mutex) != 0) {
-                WOLFSSL_MSG("Bad Init Mutex rand methods");
-                ret = BAD_MUTEX_E;
-            }
-            else {
-                initRandMethMutex = 1;
-            }
-        }
-
-        if (initRandMethMutex != 0) {
-            if (wc_LockMutex(&rand_methods_mutex) != 0) {
-                WOLFSSL_MSG("Lock rand methods mutex failed");
-                ret = BAD_MUTEX_E;
-            }
-            else {
-                if (rand_methods != NULL && rand_methods->status != NULL) {
-                    if (func == RAND_SEED) {
-                        *func_ptr = rand_methods->seed;
-                    }
-                    else if (func == RAND_BYTES) {
-                        *func_ptr = rand_methods->bytes;
-                    }
-                    else if (func == RAND_CLEANUP) {
-                        *func_ptr = rand_methods->cleanup;
-                    }
-                    else if (func == RAND_ADD) {
-                        *func_ptr = rand_methods->add;
-                    }
-                    else if (func == RAND_PSEUDORAND) {
-                        *func_ptr = rand_methods->pseudorand;
-                    }
-                    else if (func == RAND_STATUS) {
-                        *func_ptr = rand_methods->status;
-                    }
-                }
-                wc_UnLockMutex(&rand_methods_mutex);
-                ret = WOLFSSL_SUCCESS;
-            }
-        }
-
-        return ret;
+    #else
+        (void)methods;
```

Comment:
I think it makes more sense to return WOLFSSL_FAILURE if this function is not enabled.